public/js/test/mochitest/browser_dbg-breaking.js

```diff
@@ -0,0 +1,41 @@
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+// Tests the breakpoints are hit in various situations.
+
+add_task(function* () {
+  const dbg = yield initDebugger("doc-scripts.html", "scripts.html");
+  const { selectors: { getSelectedSource }, getState } = dbg;
+
+  // Make sure we can set a top-level breakpoint and it will be hit on
+  // reload.
+  yield addBreakpoint(dbg, "scripts.html", 18);
+  reload(dbg);
+  yield waitForPaused(dbg);
+  assertPausedLocation(dbg, "scripts.html", 18);
+  yield resume(dbg);
+
+  const paused = waitForPaused(dbg);
+
+  // Create an eval script that pauses itself.
+  invokeInTab("doEval")
+
+  yield paused;
```

Comment:
`waitForPaused` waits for a `paused` thread event. I was thinking that it might be racy to invoke a function that paused the debugger, and then wait for a paused event, because the `paused` event may already have been fired. However, theoretically since JS is single-threaded that event can't come through until we stop executing, so it should be fine to use `waitForPaused` after it. If we see any signs of intermittence with this test though, that's probably the problem.